state/factory/workingset_test.go

```diff
@@ -0,0 +1,161 @@
+		require.NoError(retErr)
+		require.Equal(retrdf, "test")
+	})
+
+	t.Run("Unload call error", func(t *testing.T) {
+		retrdf, retErr := ws.Unload("")
+		expectedErrors := errors.New("name : name does not exist")
+		require.EqualError(retErr, expectedErrors.Error())
+		require.Equal(retrdf, nil)
+	})
+
+	t.Run("Load Unload normal call ", func(t *testing.T) {
+		require.NoError(ws.Load("test", "test"))
+		retrdf, retErr := ws.Unload("test")
+		require.NoError(retErr)
+		require.Equal(retrdf, "test")
+	})
+}
+
+func TestProtocolDirty(t *testing.T) {
+	require := require.New(t)
+	sf, err := NewFactory(config.Default, InMemTrieOption())
+	require.NoError(err)
+	ctx := protocol.WithBlockchainCtx(
+		protocol.WithRegistry(context.Background(), protocol.NewRegistry()),
+		protocol.BlockchainCtx{Genesis: config.Default.Genesis},
+	)
+	require.NoError(sf.Start(ctx))
+	ws, err := sf.(workingSetCreator).newWorkingSet(ctx, 1)
+	require.False(ws.ProtocolDirty("test"))
+
+	require.NoError(ws.Load("test", "test"))
+	require.True(ws.ProtocolDirty("test"))
+}
+
+func TestValidateBlock(t *testing.T) {
+	require := require.New(t)
+	sf, err := NewFactory(config.Default, InMemTrieOption())
+	require.NoError(err)
+	gasLimit := testutil.TestGasLimit * 100000
+	zctx := protocol.WithBlockCtx(context.Background(),
+		protocol.BlockCtx{
+			BlockHeight: uint64(1),
+			Producer:    identityset.Address(27),
+			GasLimit:    gasLimit,
+		})
+	zctx = protocol.WithBlockchainCtx(
+		zctx,
+		protocol.BlockchainCtx{Genesis: config.Default.Genesis},
+	)
+	require.NoError(sf.Start(zctx))
+
+	t.Run("normal call", func(t *testing.T) {
+		blk := makeBlock(t, 0, hash.ZeroHash256, hash.Hash256b([]byte{65, 99, 99, 111, 117, 110, 116, 99, 117, 114, 114, 101, 110, 116, 72, 101, 105, 103, 104, 116, 1, 0, 0, 0, 0, 0, 0, 0}))
+		require.NoError(sf.Validate(zctx, blk))
+	})
+
+	t.Run("nonce error", func(t *testing.T) {
+		blk := makeBlock(t, 1, hash.ZeroHash256, hash.Hash256b([]byte{65, 99, 99, 111, 117, 110, 116, 99, 117, 114, 114, 101, 110, 116, 72, 101, 105, 103, 104, 116, 1, 0, 0, 0, 0, 0, 0, 0}))
+		require.NoError(sf.Validate(zctx, blk))
+	})
+
+	t.Run("root hash error", func(t *testing.T) {
+		blk := makeBlock(t, 0, hash.Hash256b([]byte("test")), hash.Hash256b([]byte("test")))
+		expectedErrors := errors.New("failed to validate block with workingset in factory:" +
+			" failed to verify delta state digest: delta state digest doesn't match, " +
+			"expected = 9c22ff5f21f0b81b113e63f7db6da94fedef11b2119b4088b89664fb9a3cb658, " +
+			"actual = b672c9597b38b451ea5fc53536e3fadd197b12f035f40a26122a66b8d7fa6e3f")
+		require.EqualError(sf.Validate(zctx, blk), expectedErrors.Error())
+	})
+
+	t.Run("delta state digest error", func(t *testing.T) {
+		blk := makeBlock(t, 0, hash.ZeroHash256, hash.Hash256b([]byte("test")))
+		expectedErrors := errors.New("failed to validate block with workingset in factory: " +
+			"failed to verify delta state digest: delta state digest doesn't match," +
```

Comment:
define this "delta state digest doesn't match" = ErrDeltaStateMismatch in block.go
then use Equal(block.ErrDeltaStateMismatch, errors.Cause(err))
no need to define the exact expectedErrors